BuilderDesignPattern/src/main/java/com/designPattern2/FlatBuilder.java

```diff
@@ -0,0 +1,43 @@
+package com.designPattern2;
+
+/**
+  Concrete builders implement steps defined in the common interface.
+ 
+  Unlike other creational patterns, Builder can construct unrelated products,
+  which don't have the common interface.
+ 
+  In this case we build a user manual for a car, using the same steps as we
+  built a car. This allows to produce manuals for specific car models,
+  configured with different features.
+*/
+
+public class FlatBuilder implements Builder {
+    private BuildingType type;
+    private int bedrooms;
+    private int bathrooms;
+    private boolean garden;
+
+    @Override
+    public void setBuildingType(BuildingType type) {
+        this.type = type;
+    }

```

Comment:
In what situation would the `BuildingType` for a Flat not be `BuildingType.FLAT`? It might be an idea to always set this to `FLAT` as opposed to passing it in as a parameter

Reply:
Good shout this would make sense to set the parameter at this level instead of doing it in the Director/client